pyomo/dae/tests/test_set_utils.py

```diff
@@ -0,0 +1,207 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+"""
+Unit Tests for pyomo.dae.set_utils
+"""
+import os
+from os.path import abspath, dirname
+
+from six import StringIO
+
+import pyutilib.th as unittest
+
+from pyomo.environ import *
+from pyomo.common.log import LoggingIntercept
+from pyomo.dae import *
+from pyomo.dae.set_utils import *
+from pyomo.core.kernel.component_map import ComponentMap
+
+currdir = dirname(abspath(__file__)) + os.sep
+
+
+class TestDaeSetUtils(unittest.TestCase):
```

Comment:
Could you add a few more tests for components that are also indexed by multi-dimensional Sets and components where the ContinuousSet is not always the first indexing set? Looks like you have these kinds of tests for one function but not the other. I've seen edge cases and bugs crop up with multi-dimensional sets and set ordering when implementing similar functionality.

Reply:
I believe I have implemented satisfactory tests.